src/lexer/decodeHTML.ts

```diff
@@ -2163,10 +2163,11 @@ export function decodeHTMLStrict(text: string): string {
   return text.replace(/&(?:[a-zA-Z]+|#[xX][\da-fA-F]+|#\d+);/g, (key) => {
     if (key.charAt(1) === '#') {
       const secondChar = key.charAt(2);
-      if (secondChar === 'X' || secondChar === 'x') {
-        return decodeCodePoint(parseInt(key.substr(3), 16));
-      }
-      return decodeCodePoint(parseInt(key.substr(2), 10));
+    const codePoint =
+      secondChar === 'X' || secondChar === 'x'
```

Comment:
what is 'x' ?  Note that this can be simplified and force to lowercase `secondChar | 32 === codePoint`. If no codepoint yo u can do the same trick with 0x20